app-sdk/src/lib.rs

```diff
@@ -125,3 +125,29 @@ macro_rules! bootstrap {
         }
     };
 }
+
+#[no_mangle]
+#[inline(never)]
+#[cfg(target_arch = "riscv32")]
+pub unsafe extern "C" fn memcpy(dest: *mut u8, src: *const u8, n: usize) -> *mut u8 {
+    Ecall::sys_memcpy(dest, src, n)
+}
+
+#[no_mangle]
+#[inline(never)]
+#[cfg(target_arch = "riscv32")]
+pub unsafe extern "C" fn memset(dest: *mut u8, ch: i32, n: usize) -> *mut u8 {
+    Ecall::sys_memset(dest, ch, n)
+}
+
+// TODO: do we want to implement and replace memcpy/memset on native targets?
+//       It isn't really useful, but it could be worth reducing the difference
+//       between targets (or properly document the differences).
```

Comment:
I think it's best to override it on as low level as possible and don't worry about "difference" because the behavior will be the same.